src/dataAccessLayer.ts

```diff
@@ -67,4 +98,44 @@ export class DataAccessLayer {
     public async clearStatesAsync(): Promise<void> {
         await this.stateRepository.clearStatesAsync();
     }
+
+    public async getTransactionsAsync(): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsAsync();
+    }
+
+    public async getTransactionsByAmountAsync(amount: number): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsByAmountAsync(amount);
+    }
+
+    public async getTransactionsByDateAsync(timestamp: number): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsByDateAsync(timestamp);
+    }
+
+    public async getTransactionsByPeriodAsync(beginTimestamp: number, endTimestamp: number): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsByPeriodAsync(beginTimestamp, endTimestamp);
+    }
+
+    public async getTransactionsByRecipientAsync(recipient: string): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsByRecipientAsync(recipient);
+    }
+
+    public async getTransactionsBySenderAsync(sender: string): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsBySenderAsync(sender);
+    }
+
+    public async getTransactionsBySignatureAsync(signature: string): Promise<Transaction[]> {
+        return await this.transactionRepository.getTransactionsBySignatureAsync(signature);
+    }
+
+    public async getTransactionsByRouteAsync(sender: string, recipient: string): Promise<Transaction[]> {
```

Comment:
Could you please clarify the naming of the method? What is `Route`?

Reply:
The route between sender and recipient. Agree that this isn't clear from the function name. This will be changed to `getTransactionsBySenderRecipientAsync()`.